File: samples/java/src/main/java/co/windly/limbosample/presentation/main/MainActivity.java
package co.windly.limbosample.presentation.main;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import butterknife.BindView;
import butterknife.ButterKnife;
import co.windly.limbo.fragment.base.LimboFragment;
import co.windly.limbosample.R;
import co.windly.limbosample.presentation.base.activity.fragment.BaseFragmentActivity;
import co.windly.limbosample.presentation.main.home.HomeFragment;
import co.windly.limbosample.presentation.main.settings.SettingsFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import javax.inject.Inject;

public class MainActivity extends BaseFragmentActivity<MainView, MainPresenter> implements MainView {

  //region Intent

  public static Intent createIntent(@NonNull Context context) {
    return new Intent(context, MainActivity.class);
  }

  //endregion

  //region Ui

  @Override
  protected int getLayout() {
    return R.layout.activity_main;
  }

  //endregion

  //region Presenter

  @Inject
  protected MainPresenter mainPresenter;

  @NonNull
  @Override
  public MainPresenter createPresenter() {
    return mainPresenter;
  }

  //endregion

  //region Navigation

  @BindView(R.id.bottomNavigation)
  BottomNavigationView bottomNavigationView;

  private void initializeBottomNavigation() {

    // Configure listener.
    bottomNavigationView.setOnNavigationItemSelectedListener(item -> {

      // Retrieve item id.
      final int itemId = item.getItemId();

      // Navigate to selected item.
      switch (itemId) {
        case R.id.action_home:
          navigateToHomeView();
          break;
        case R.id.action_settings:
          navigateToSettingsView();
          break;
      }

      return true;
    });
  }

  @Override
  public void navigateToHomeView() {

    // Prepare fragment.
    final LimboFragment fragment = HomeFragment.createInstance();

    // Replace current fragment.
    getSupportDelegate()
      .replaceFragment(fragment, false);
  }

  @Override
  public void navigateToSettingsView() {

    // Prepare fragment.
    final LimboFragment fragment = SettingsFragment.createInstance();

    // Replace current fragment.
    getSupportDelegate()
      .replaceFragment(fragment, false);
  }

  //endregion

  //region Lifecycle

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    // Bind views.
    ButterKnife.bind(this);

    // Initialize bottom navigation.
    initializeBottomNavigation();

    // Load fragment.
    loadHomeFragment();
  }

  //endregion

  //region Fragment

  private void loadHomeFragment() {

    // Check whether fragment is already added.
    final LimboFragment fragment = findFragment(HomeFragment.class);

    // In case if not added - load.
    if (fragment == null) {
      loadRootFragment(R.id.fragmentContainer, HomeFragment.createInstance());
    }
  }

  //endregion
}

File: samples/kotlin/src/main/java/co/windly/limbosample/presentation/main/MainActivity.kt
package co.windly.limbosample.presentation.main

import android.content.Context
import android.content.Intent
import android.os.Bundle
import co.windly.limbosample.R
import co.windly.limbosample.presentation.base.activity.base.BaseActivity
import co.windly.limbosample.presentation.main.home.HomeFragment
import co.windly.limbosample.presentation.main.settings.SettingsFragment
import kotlinx.android.synthetic.main.activity_main.bottomNavigation

class MainActivity : BaseActivity<MainView, MainPresenter>(), MainView {

  //region Activity

  companion object {

    fun createIntent(context: Context): Intent = Intent(context, MainActivity::class.java)
  }

  //endregion

  //region Ui

  override val layout: Int
    get() = R.layout.activity_main

  //endregion

  //region Navigation

  private fun initializeBottomNavigation() {

    // Configure listener.
    bottomNavigation.setOnNavigationItemSelectedListener {

      // Navigate to selected item.
      when (it.itemId) {
        R.id.action_home -> navigateToHomeView()
        R.id.action_settings -> navigateToSettingsView()
      }

      true
    }
  }

  override fun navigateToHomeView() {

    // Prepare fragment.
    val fragment = HomeFragment.createInstance()

    // Replace current fragment.
    supportDelegate
      .replaceFragment(fragment, false)
  }

  override fun navigateToSettingsView() {

    // Prepare fragment.
    val fragment = SettingsFragment.createInstance()

    // Replace current fragment.
    supportDelegate
      .replaceFragment(fragment, false)
  }

  //endregion

  //region Lifecycle

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    // Initialize bottom navigation.
    initializeBottomNavigation()

    // Load fragment.
    loadHomeFragment()
  }

  //endregion

  //region Fragment

  private fun loadHomeFragment() {

    // Check whether fragment is already added.
    val fragment = findFragment(HomeFragment::class.java)

    // In case if not added - load.
    if (fragment == null) {
      loadRootFragment(R.id.fragmentContainer, HomeFragment.createInstance())
    }
  }

  //endregion
}
